extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,19 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
-
       this.runOrSchedule_(() => {
         visibilityFor(this.win_).then(visibility => {
-          visibility.listenOnce(config['visibilitySpec'], vars => {
+          visibility.listenOnce(spec, vars => {
+            if (spec['selector']) {
+              const el = this.win_.document.getElementById(spec['selector']
+                .slice(1));
+              Object.assign(vars, this.extractAnalyticsVariables_(el));
```

Comment:
http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign says that it is not supported IE11 and below, Safari 8 and below. But Babel and closure support it. What can we use here instead? Is a for-loop the best alternative if `.assign` is undefined?